routes/products.js

```diff
@@ -0,0 +1,57 @@
+import express from 'express';
+import productController, { EVENTS } from '../controllers/productController';
+
+
+const routes = express.Router();
+const mainRout = '/products';
+
+routes.get(`${mainRout}`, (req, res, next) => {
+    productController.getProducts();
+    productController.on(EVENTS.getProducts, (products) => {
+        res.json(products);
+    });
+
+    productController.on('error', err => {
+        console.error('GET products: ', err);
+        res.sendStatus(500);
+    });
+});
+
+routes.get(`${mainRout}/:id`, (req, res, next) => {
+    productController.getProductId(req.params.id);
+    productController.on(EVENTS.getProductId, (products) => {
+        res.json(products);
+    });
+
+    productController.on('error', err => {
+        console.error('GET product id: ', err);
+        res.sendStatus(500);
+    });
+});
+
+routes.get(`${mainRout}/:id/reviews`, (req, res, next) => {
+    productController.getReviews(req.params.id);
+    productController.on(EVENTS.reviews, (products) => {
+        res.json(products);
+    });
+
+    productController.on('error', err => {
+        console.error('GET product id: ', err);
+        res.sendStatus(500);
+    });
+});
+
+routes.post(`${mainRout}/`, (req, res, next) => {
```

Comment:
When I send `post` request without any body, server doesn't show error response. And in the same time result of `/products` endpoint is not changed. It looks strange.